airbyte-integrations/connectors/source-stripe/unit_tests/test_streams.py

```diff
@@ -2,180 +2,178 @@
                         "data": [
                             {
-                                "id": "il_1",
-                                "object": "line_item",
+                                "id": "cs_1",
+                                "object": "card",
                             },
                             {
-                                "id": "il_2",
-                                "object": "line_item",
+                                "id": "cs_2",
+                                "object": "bank_account",
                             },
                         ],
                         "has_more": True,
                         "object": "list",
-                        "total_count": 3,
-                        "url": "/v1/invoices/in_1KD6OVIEn5WyEQxn9xuASHsD/lines",
+                        "total_count": 4,
+                        "url": "/v1/customers/cus_HezytZRkaQJC8W/sources",
                     },
                 }
             ],
         },
-    )
-
-    # Second pagination request to main stream
-    requests_mock.get(
-        "https://api.stripe.com/v1/invoices/in_1KD6OVIEn5WyEQxn9xuASHsD/lines",
-        json={
+        "https://api.stripe.com/v1/customers/cus_HezytZRkaQJC8W/sources?object=bank_account&starting_after=cs_2": {
             "data": [
                 {
-                    "id": "il_3",
-                    "object": "line_item",
+                    "id": "cs_3",
+                    "object": "card",
+                },
+                {
+                    "id": "cs_4",
+                    "object": "bank_account",
                 },
             ],
             "has_more": False,
             "object": "list",
-            "total_count": 3,
-            "url": "/v1/invoices/in_1KD6OVIEn5WyEQxn9xuASHsD/lines",
+            "total_count": 4,
+            "url": "/v1/customers/cus_HezytZRkaQJC8W/sources",
         },
-    )
+    },
+    "bank_accounts",
+    [
+        {"id": "cs_2", "object": "bank_account", "updated": 1692802815},
+        {"id": "cs_4", "object": "bank_account", "updated": 1692802815},
+    ],
+    "full_refresh",
+    {},
+)
 
-    # make start date a recent date so there's just one slice in a parent stream
-    stream_args["start_date"] = pendulum.today().subtract(days=3).int_timestamp
-    parent_stream = invoices(stream_args)
-    stream = invoice_line_items(stream_args, parent_stream=parent_stream)
-    records = []
 
-    for slice_ in stream.stream_slices(sync_mode="full_refresh"):
-        records.extend(stream.read_records(sync_mode="full_refresh", stream_slice=slice_))
-    assert list(records) == [
-        {"id": "il_1", "invoice_id": "in_1KD6OVIEn5WyEQxn9xuASHsD", "object": "line_item"},
-        {"id": "il_2", "invoice_id": "in_1KD6OVIEn5WyEQxn9xuASHsD", "object": "line_item"},
-        {"id": "il_3", "invoice_id": "in_1KD6OVIEn5WyEQxn9xuASHsD", "object": "line_item"},
-    ]
+bank_accounts_incremental_test_case = (
```

Comment:
This tests `Data not filtered when using the event-based incremental sync mode` 👍